### DISCUSSION TITLE
Docker - difficulty installing mysql8

### DISCUSSION THREAD

**Discussion by fabiodurao** [SCADA-LTS/Scada-LTS#2616] [General]
Hello Kamil, I tried to run docker on my linux machine, where I always tested docker images and this time I was unsuccessful. I followed the docker-compose tutorial, I even bothered to get the most up-to-date docker-compose.yml here on github and deleted all containers and previous images to avoid any problems

1.I downloaded docker-compose.yml
1.1 I changed the output port to 8081 so as not to conflict with another tomcat that I have installed (in the same way that I always did to test the Scada-LTS images);
2.First you need to start the database:
docker-compose up database
3.When the database starts then:
docker-compose up ladders

and got the feedback below
![image](https://github.com/SCADA-LTS/Scada-LTS/assets/38673539/ed9205ce-450e-4b58-a104-28ed69194ce3)
![image](https://github.com/SCADA-LTS/Scada-LTS/assets/38673539/1fe5fe06-52c3-4e02-9a27-2e9e0fe98a56)

Am I missing something?

**Comment by Limraj** [comment]
Hi @fabiodurao,
Are you running this file in a location without db_data etc folders?
The place where you have the docker-compose.yml file should be clean. I am not able to determine what you were running there before, etc. If something does not work, you have to try cleanly. Check if port 3306 is free.

Maybe you were running it with a different linux user before, and now you're running it with root privileges. Then any other user can have limited access to these created folders.

Changing the port for tomcat/scadalts shouldn't matter here. 
Indicates a problem creating a container for mysql. (edit: With the start of the service, so it probably created a container.)
Didn't you check for the default port 8080?

Regards,
Kamil Jarmusik

**Comment by fabiodurao** [comment]
I was checking what you told me about the folder, I changed the file to an empty folder, before that I deleted a folder that existed with the name "docker", but it was empty, after deleting the folder and changing the docker-compose file .yml to another folder and also delete the container, I ran "docker-compose up database" and it gave the same return.
I searched for ports with nmap, lsof, netstat and I didn't find anything about port 3306 being used, but just in case, how should the syntax of the .yml file be to change to port 3307. I believe that I should not change all numbers 3306 that you find in the file by 3307, I imagine that there are ports inside the container and others outside, but I don't quite understand, could you give me a modified template so I can test this, please?

**Comment by Limraj** [comment]
@fabiodurao I'm talking about where the docker-compose.yml file is empty outside of this file, can you do it like this?

edit:
1. Take a screenshot of what you have in the folder with the docker-compose.yml file before and after running.

2. Next Try restarting the system/machine:
https://serverfault.com/questions/738773/docker-failed-to-add-the-pair-interfaces-operation-not-supported

3. If it still doesn't work, make the user you are running from the owner of the folder and the docker-compose.yml file, you can also grant all permissions to this file and folder.

4. How did you delete the container? If deleting the container doesn't work, you need to delete the image and download it again.

5. If it still doesn't work and you don't have the important data in the docker you can call the command:
`docker system prune -a --volumes --force`
https://docs.docker.com/engine/reference/commandline/system_prune/

6. Next try delete the database service and try to run tomcat itself, remove property 'command:' and 'links:' in scadalts service. 
If you delete something in this file, download a new one, docker is sensitive to spaces and other whitespace.

**Comment by fabiodurao** [comment]
Success in step 2!

1. The folder was empty before and after running docker-compose up database;

2. I restarted, deleted all container and images and ran again and now it went up and folders were created inside the folder where the .yml is.

![image](https://github.com/SCADA-LTS/Scada-LTS/assets/38673539/c2c53147-2be6-4850-9f24-28fc51c06781)


Many thanks for the support!

**Comment by Limraj** [comment]
Okey, iam closed.
